src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -462,23 +460,38 @@ private List<DropStack> GenerateDropStacks(Character killer)
 				var lootingRate = 1f + lootingChance * 0.001f;
 				adjustedDropChance *= lootingRate;
 
-				// Items with a chance of >0.5% always drop on super drop.
-				// We'll use the original drop chance for this check to
-				// get a constistent drop behavior
-				var guaranteedDrop = superDropLevel > 0 && originalDropChance > 0.5f;
-				if (!guaranteedDrop)
+				// Calculate Enhanced drops for super mobs
+				var isSuperMob = this.TryGetSuperMob(out var superMobType);
+				var superMobRerolls = 0;
+				var superMobGuaranteedItemDrop = false;
+				var superMobIncreaseMoney = 0;
+				var superMobMoneyStacks = 0;
+				if (isSuperMob)
 				{
-					var superAdjustedDropChance = adjustedDropChance;
-
-					// Increase drop chance for super drops
-					if (superDropLevel == 1)
-						superAdjustedDropChance += 1;
-					else if (superDropLevel == 2)
-						superAdjustedDropChance += 10;
-
-					var dropSuccess = rnd.NextDouble() < superAdjustedDropChance / 100f;
-					if (!dropSuccess)
-						continue;
+					switch (superMobType)
+					{
+						case SuperMobType.Silver:
+							superMobRerolls = worldConf.SilverJackpotRolls;
+							superMobGuaranteedItemDrop = originalDropChance > worldConf.SilverJackpotGuaranteedItemThreshold;
+							superMobIncreaseMoney = worldConf.SilverJackpotRolls;
+							superMobMoneyStacks = RandomProvider.Next(40, 50);
```

Comment:
I would generally recommend using `RandomProvider.Get().Next` over `RandomProvider.Next`, because the latter needs to lock a `Random` instance to produce a number. The former creates one instance per thread that can then be used unobstructed, and any additional calls from the same thread will return that already existing instance.

Though in this specific case you already have a `Random` instance in the variable `rnd`, which you could use directly.